src/math.ts

```diff
@@ -82,6 +83,184 @@ export function adjustXYWithRotation(
   return { x, y };
 }
 
+export function calculateResizedPosition(
```

Comment:
For the same reason, we should have this function in `src/element/`. It it too big to embed this in `resizeElements.ts`?
It would be nice if we could extract pure math code out of this.

Reply:
I think I could extract something from here and also reduce the code by split the switch and reuse some part of the code which are used for more than one side...

Reply:
Sounds great.

Reply:
> Great work! It's a bit unfortunate that the code gets bigger. I was hoping the number of lines won't change much. There seems some room for improvement. I can work on that unless you are very happy with polishing.

Anyways, Happy if you want to polish it! As I'm starting working now until 17.00 pm ( London Time ), So, I could not complete before tonight, and people waiting for that PR, so I think is sensible that you polish it if you want :)